### PR TITLE
try atom_indices

### PR DESCRIPTION
**Pull request by hainm**
e.g.

```python
atom_indices = [3, 8, 9, 12]
w.add_representation('licorice', selection=atom_indices, color='red')

# or 
w.licorice(selection=atom_indices, color='red')
```

### CODE REVIEW

**Comment by hainm** [comment]
I have not tested yet. But what do you think?

File: nglview/utils.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+from six import string_types
+
+def seq_to_string(seq):
+    """e.g. convert [1, 3, 5] to "@1,3,5
+    """
+    if isinstance(seq, string_types):
+        return seq
+    else:
+        # assume 1D array
+        return "@" + ",".join(seq)
```
Comment (swails) [inline nglview/utils.py]: This won't work when `seq` is a sequence of non-strings:

```python
>>> '@' + ','.join([1, 2, 3, 4, 5])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: sequence item 0: expected str instance, int found
```

Comment (hainm) [inline nglview/utils.py]: Thanks. Poor habit of not writing test. I will update.

**Comment by arose** [comment]
I like this. The next ngl version will support bitsets as a source for selections which will speed up selecting large atom index arrays.

**Comment by hainm** [comment]
@arose 

I've just updated this PR
- fix string convert as suggested by @swails 
```python
>>> '@' + ','.join([1, 2, 3, 4, 5])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: sequence item 0: expected str instance, int found
```

- add test for that + add test framework (nose) too

- add six module for PY2, 3 compat.

Questions:
For atom index selection, does ngl use 0-based index or 1-based? 

```python
[1, 2, 3] should be @1,2,3 or @2,3,4?
```

**Comment by arose** [comment]
> For atom index selection, does ngl use 0-based index or 1-based?

zero-based

**Comment by hainm** [comment]
ok, it's good to merge now. 

(We should note the users that (0-based) to since AMBER use 1-based with similar syntax :D).

**Comment by arose** [comment]
Is including six.py really necessary? It is just for the string type test, right? Can we handle that ourselves? Can we import a package?

**Comment by swails** [comment]
+1 for bundling `six.py`.  It's lightweight, not changing fast (and updates are basically never required, anyway), and is convenient to use for Py2-Py3 compatible code bases.

Basically, any time you use `range`, `zip`, `StringIO`, `dict.items`/`dict.iteritems`, you should include it from `six`.  If it's going to be required, anyway, it's just as easy to bundle it explicitly as it is to require users to install the standalone copy IMO.  It's *certainly* needed of you use metaclasses.

Just pulling out the pieces you need or want to use at the current time means that every time you want to use something else (or find somewhere that using the `moves` from `six` is better for Py2), you need to pull in that extra bit of functionality into whatever you wrote in place of `six`.  And `six` is so well-tested and so widely-used that you wouldn't want to do something *else*, really.

Just my 2c.

**Comment by hainm** [comment]
@arose What's Jason said is what I wanted to bundle six.py in nglview. It's good for future development. It's light though.

**Comment by arose** [comment]
I am convinced to use six.py, thanks guys. However I would prefer importing the package instead of bundling.

@swails did I understand you right, bundling is needed if you use metaclasses?

**Comment by arose** [comment]
@hainm I would be great if you could also specify `tests_require` and `test_suite` in `setup.py`. Thanks!

**Comment by swails** [comment]
bundling is never required over simply importing and requiring users to install it first.  If you have a `requirements.txt` file that pip automatically goes out and gets, then it's probably fine to not bundle it and simply make it a requirement.  Six of one, half dozen of another -- it's up to personal preferences.  Some would prefer *not* to bundle, and that's fine, while others would prefer to bundle (like me), and that's also fine.

> @swails did I understand you right, bundling is needed if you use metaclasses?

No -- `six.py` is required if you use metaclasses.  As opposed to simply doing your own 2/3 compatibility module with just the stuff you need.  The syntax for metaclasses in Python 2 and Python 3 are different and incompatible.  six includes a class decorator that implements the full Python 2 functionality of metaclasses (Python 3 can do more with them), but the decorator works in both Python 2 and Python 3.

**Comment by arose** [comment]
Thank you very much for the explanation @swails 

In the case of six.py I prefer not to bundle but am ok with it if anyone feels strongly.

**Comment by hainm** [comment]
I have no problem with using it externally, just a matter of taste. I will remove it.

**Comment by hainm** [comment]
updated

- remove six.py
- add test_suite + test_require

Q: @arose do you know why the test failed with py3? ImportError for six (although I specify in setup.py)

**Comment by hainm** [comment]
uhm, new problem.

Do we really need to add test_require and test_suite. It make things more complicated for testing.

I tried to `python setup.py test` in py3.4
got 

```bash
Searching for entrypoints
Reading https://pypi.python.org/simple/entrypoints/
No local packages or download links found for entrypoints
error: Could not find suitable distribution for Requirement.parse('entrypoints')
```

**Comment by hainm** [comment]
nvm. I thought "simple" is "simpletraj". lol.

**Comment by hainm** [comment]
ok. good for review (again) now.

**Comment by hainm** [comment]
merged. I have so many branches so I need to merge some of them if look ok (so I don't need to spend time on dealing with merge conflict).

**Pull request merged**